ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-Patient.xsl

```diff
@@ -0,0 +1,360 @@
+                <nm:resource type="Patient">
+                    <!-- What would be the use case? -->
+                    <xsl:if test="./@id">
+                        <nm:ada-id>
+                            <xsl:value-of select="@id"/>
+                        </nm:ada-id>
+                    </xsl:if>
+                    <nm:group-key>
+                        <xsl:value-of select="current-grouping-key()"/>
+                    </nm:group-key>
+                    <!-- Unless overruled, fullUrl is always a uuid? -->
+                    <xsl:variable name="fullUrl">
+                        <xsl:value-of select="nf:get-fhir-uuid(current-group()[1])"/>
+                    </xsl:variable>
+                    <nm:full-url>
+                        <xsl:value-of select="$fullUrl"/>
+                    </nm:full-url>
+                    <xsl:if test="$referById">
+                        <nm:logical-id>
+                            <xsl:choose>
+                                <!-- All $uuid parts commented out, see above. -->
+                                <!-- Tries to match patient to token -->
+                                <xsl:when test="
+                                        (:not($uuid) and :)
+                                        string-length(nf:get-resourceid-from-token(.)) gt 0">
+                                    <xsl:value-of select="nf:get-resourceid-from-token(.)"/>
+                                </xsl:when>
+                                <xsl:when test="
+                                        (:not($uuid) and :)
+                                        naamgegevens[1]//*[not(name() = 'naamgebruik')]/@value">
+                                    <xsl:value-of select="upper-case(nf:removeSpecialCharacters(normalize-space(string-join(naamgegevens[1]//*[not(name() = 'naamgebruik')]/@value, ' '))))"/>
+                                </xsl:when>
+                                <xsl:otherwise>
+                                    <xsl:value-of select="nf:removeSpecialCharacters(replace($fullUrl, 'urn:[^i]*id:', ''))"/>
+                                </xsl:otherwise>
+                            </xsl:choose>
+                        </nm:logical-id>
+                    </xsl:if>
+                    <nm:reference-display>
+                        <xsl:value-of select="current-group()[1]/normalize-space(string-join(.//naamgegevens[1]//*[not(name() = 'naamgebruik')]/@value | name_information[1]//*[not(name() = 'name_usage')]/@value, ' '))"/>
+                    </nm:reference-display>
+                    <!-- With all data above, what is the exact use case of having the rendered FHIR resource here? -->
```

Comment:
Dat is de hamvraag. Wil je per se eerst alle Patient-resources renderen, dan alle resources die naar Patient verwijzen, dan alle alle resources die daar weer naar verwijzen etc? Of is het makkelijker om gewoon eerst alle metadata op te stellen en daarmee de resources te renderen. Ik voel meer voor optie 2, maar ik kan niet overzien wat dat bv. betekent voor debugging.

Reply:
Ik zou het ook op manier 2 in willen steken. Als we het daarmee redden dan is dat mooier. Blijkt dat we het toch niet redden dan kunnen we altijd nog de andere richting in.